mkosi/resources/dracut_unified_kernel_install.sh

```diff
@@ -15,7 +15,13 @@ fi
 PREFIX=$(dirname $(dirname "$BOOT_DIR_ABS"))
 
 # Pick a default prefix name for the unified kernel binary
-if [[ -z "$IMAGE_ID" ]] ; then
+if [[ -n "$IMAGE_ID" ]] ; then
+    if [[ -n "$IMAGE_VERSION" ]]; then
+        PARTLABEL="${IMAGE_ID}_${IMAGE_VERSION}"
+    else
+        PARTLABEL="${IMAGE_ID}"
+    fi
+else
     IMAGE_ID=linux
```

Comment:
mmh `PARTLABEL` is not set in this branch, no? shouldn't this be changed to `PARTLABEL=linux` ?

Reply:
no. the idea is if the PARTLABEL env var is set we use it to rference the root fs. But if we have no IMAGE_ID then the partition label is not supposed to be an "id" but just some string that nice for humans to read, and apparently people of no intention to make the image or partitions identifiable with it. Thus in that case we want to rely on systemd-gpt-auto-generator's logic of automatically finding the roots for you, by looking for a any suitable partition on the disk with the ESP on it.

Reply:
But then PARTLABEL should be set as an empty string at least, no? Otherwise it's used while undefined, which is not great and would make shellcheck angry:

```
BOOT_OPTIONS="${BOOT_OPTIONS} root=PARTLABEL=${PARTLABEL}"
```

Reply:
actually that's not true, shellcheck is happy with -n $undefined, so ignore me